### PR TITLE
Unify beans

### PR DESCRIPTION
**Pull request by lauraluiz**
@schleichardt please review. As I stumbled over the cart stuff, I preferred to leave it here and continue with the task in a future PR. It became already too many changes.

### CODE REVIEW

File: common/app/com/commercetools/sunrise/shoppingcart/CartBean.java [hunk 1]
```diff
@@ -1,50 +1,18 @@
...
-import com.commercetools.sunrise.common.ctp.ProductDataConfig;
 import com.commercetools.sunrise.common.models.AddressBean;
-import com.commercetools.sunrise.common.utils.MoneyContext;
-import io.sphere.sdk.carts.CartLike;
-import io.sphere.sdk.carts.LineItem;
-import io.sphere.sdk.orders.Order;
-import com.commercetools.sunrise.common.reverserouter.ProductReverseRouter;
 
-import java.time.format.DateTimeFormatter;
+public class CartBean extends MiniCartBean {
```
Comment (schleichardt) [inline common/app/com/commercetools/sunrise/shoppingcart/CartBean.java]: no "no go", but inheritance makes it sometimes very hard to work with stuff...

File: shopping-cart/app/com/commercetools/sunrise/shoppingcart/common/SunriseFrameworkCartController.java [hunk 2]
```diff
@@ -117,14 +114,15 @@ private void postInit() {
 
     protected CompletionStage<Cart> applySideEffects(@Nonnull Cart cart) {
         final Http.Session session = session();
-        return updateCartWithUserPreferences(cart, userContext()).thenApply(updatedCart -> {
-            CartSessionUtils.overwriteCartSessionData(updatedCart, session, userContext(), productReverseRouter);
-            return updatedCart;
-        })
+        return updateCartWithUserPreferences(cart, userContext())
                 .thenApply(updatedCart -> {
+                    final MiniCartBeanFactory miniCartBeanFactory = injector().getInstance(MiniCartBeanFactory.class);
```
Comment (lauraluiz) [inline shopping-cart/app/com/commercetools/sunrise/shoppingcart/common/SunriseFrameworkCartController.java]: This is actually the only thing added here, the rest it's just a bit of (involuntary) formatting.

File: common/app/com/commercetools/sunrise/shoppingcart/OrderBeanFactory.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    @Inject
+    private LineItemExtendedBeanFactory lineItemExtendedBeanFactory;
+
+    public OrderBean create(final Order order) {
+        final OrderBean bean = new OrderBean();
+        initialize(bean, order);
+        return bean;
+    }
+
+    protected final void initialize(final OrderBean bean, final Order order) {
```
Comment (schleichardt) [inline common/app/com/commercetools/sunrise/shoppingcart/OrderBeanFactory.java]: here is maybe duplication, the cart has mostly the same fields

**Comment by schleichardt** [comment]
ok, but see the comments

**Pull request merged**